force-app/main/default/classes/TestRecordsFactory.cls

```diff
@@ -0,0 +1,97 @@
+@isTest
+public class TestRecordsFactory {
+  public static void insertUser(String profileName) {
```

Comment:
Few suggestions

`TestRecordsFactory -> TestDataFactory`

insertUser -> createUser. It should return new User(...);
To bulkify it, pass profileId instead of profileName.

Do the same for every other method:
`insertPositions -> createPosition`
It should create new instance of Position, and if you need multiple Positions, you will call createPosition in for-loop, and add it to list of positions which you should insert later.

Params that you pass should be pretty simple, and if it's possible, it should be used as value for field.
`Title__c = keyWord + i `-> replace keyWord with title variable, and update this param outside
`Title__c = title`

```
List<Position__c> positions = new List<Position__c>();
for (Integer i = 0; i <= numberOfTestedPositions; i++) {
    positions.add(TestRecordsFactory.createPositions('Test Title ' + i);
}
insert positions;
```

Reply:
Feel free to ignore the implementation for the above change request. It's just for knowledge.